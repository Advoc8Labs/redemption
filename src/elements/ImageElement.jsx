import React from 'react';

function ImageElement({ element }) {
    const styles = element.applyStyles;
    return (
        <div
            css={{
                display: 'flex',
                ...styles.getLayout(),
                ...styles.getTarget('container')
            }}
        >
            <img
                src={element.source_url}
                alt='Form Image'
                css={{
                    objectFit: 'contain',
                    ...styles.getTarget('image')
                }}
            />
        </div>
    );
}

export default ImageElement;
