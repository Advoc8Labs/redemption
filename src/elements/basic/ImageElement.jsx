import React, { useMemo } from 'react';

function applyImageStyles(element, applyStyles) {
  applyStyles.addTargets('image');
  applyStyles.applyMargin('image');
  applyStyles.applyWidth('image');

  if (element.styles.line_height) {
    applyStyles.apply('text', 'line_height', (a) => ({
      lineHeight: `${a}px`
    }));
  }

  return applyStyles;
}

function ImageElement({ element, applyStyles, elementProps = {}, children }) {
  const styles = useMemo(() => applyImageStyles(element, applyStyles), [
    applyStyles
  ]);
  return (
    <div css={{ ...styles.getTarget('image'), position: 'relative' }}>
      <img
        src={element.properties.source_image}
        alt='Form Image'
        css={{
          objectFit: 'contain',
          width: '100%'
        }}
        {...elementProps}
      />
      {children}
    </div>
  );
}

export default ImageElement;
