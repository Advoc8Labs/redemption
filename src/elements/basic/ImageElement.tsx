import React, { useMemo } from 'react';

const PLACEHOLDER_IMAGE =
  'https://feathery.s3.us-west-1.amazonaws.com/theme-image-preview.png';

function applyImageStyles(element: any, responsiveStyles: any) {
  responsiveStyles.addTargets('imageContainer', 'image');
  responsiveStyles.applyWidth('imageContainer');
  responsiveStyles.applyCorners('imageContainer');
  responsiveStyles.applyCorners('image');
  return responsiveStyles;
}

function ImageElement({
  element,
  responsiveStyles,
  elementProps = {},
  children
}: any) {
  const styles = useMemo(
    () => applyImageStyles(element, responsiveStyles),
    [responsiveStyles]
  );
  return (
    <div
      css={{
        ...styles.getTarget('imageContainer'),
        maxHeight: '100%',
        position: 'relative',
        display: 'flex',
        alignItems: 'center'
      }}
    >
      {children}
      <img
        src={element.properties.source_image || PLACEHOLDER_IMAGE}
        alt='Form Image'
        css={{
          objectFit: 'contain',
          width: '100%',
          maxHeight: '100%',
          ...styles.getTarget('image')
        }}
        {...elementProps}
      />
    </div>
  );
}

export default ImageElement;
