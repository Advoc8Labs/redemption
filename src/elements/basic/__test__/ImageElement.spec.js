import React from 'react';
import { create, act } from 'react-test-renderer';
import ImageElement from '../ImageElement';
import ApplyStyles from '../../styles';

describe('ImageElement', () => {
  it('renders an Image element', async () => {
    // Arrange
    const props = {
      element: {
        properties: { source_image: '' },
        column_index: 0,
        row_index: 0,
        column_index_end: 0,
        row_index_end: 0,
        styles: {
          border_top_color: '2954af',
          border_right_color: '2954af',
          border_bottom_color: '2954af',
          border_left_color: '2954af',
          background_color: '2954af',
          layout: 'center'
        },
        mobile_styles: {}
      }
    };
    props.applyStyles = new ApplyStyles(props.element, []);

    // Act
    let image;
    act(() => {
      image = create(<ImageElement {...props} />);
    });

    // Assert
    expect(image).toMatchSnapshot();
  });
});
