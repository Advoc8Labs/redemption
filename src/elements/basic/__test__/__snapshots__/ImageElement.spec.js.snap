// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageElement renders an Image element 1`] = `
<img
  alt="Form Image"
  className="css-jyz4m3-ImageElement"
  src=""
/>
`;
